### PR TITLE
feat(annotator): 🚀 introduce circle annotator

### PR DESCRIPTION
**Pull request by onuralpszr**
# Description

We have box annotator and for support different shape I added circle annotator

List any dependencies that are required for this change.

## Type of change

Please delete options that are not relevant.

-   [X] New feature (non-breaking change which adds functionality)

### How has this change been tested, please provide a testcase or example of how you tested the change?

```python

from ultralytics import YOLO
import supervision as sv
import cv2

circle_annotator = sv.CircleAnnotator()

model = YOLO("yolov8n.pt") 
results = model("bus.jpg")
detections = sv.Detections.from_ultralytics(results[0])
frame = cv2.imread("bus.jpg")
annotated_frame = circle_annotator.annotate(
                scene=frame.copy(), 
                detections=detections, 
            )

cv2.imshow("frame", annotated_frame)
cv2.waitKey(0)

```

### CODE REVIEW

**Comment by SkalskiP** [comment]
Hi, @onuralpszr 👋🏻! Would you be so kind as to update this [colab](https://colab.research.google.com/drive/1iso4sYNNERe6KqH6zUixF7XwsapbCgzZ?usp=sharing) or duplicate it and show case how annotator works? It would make my life a lot simpler! 🙏🏻

**Comment by onuralpszr** [comment]
Of course

**Comment by SkalskiP** [comment]
Thank you! 🙏🏻 Let me know once you are done ;)

**Comment by onuralpszr** [comment]
@SkalskiP  https://colab.research.google.com/drive/1y5lbb0dZwy1rFcu6KHtLYmkeJ1WsVF7J?usp=sharing

**Comment by SkalskiP** [comment]
@onuralpszr, you know you don't need to work on your fork. You can create branches directly in supervision.

**Comment by onuralpszr** [comment]
> @onuralpszr, you know you don't need to work on your fork. You can create branches directly in supervision.

Habits :=)

File: supervision/annotators/core.py [hunk 1]
```diff
@@ -315,6 +316,86 @@ def annotate(self, scene: np.ndarray, detections: Detections) -> np.ndarray:
...
+
+            >>> image = ...
+            >>> detections = sv.Detections(...)
+
+            >>> circle_annotator = sv.CircleAnnotator()
+            >>> annotated_frame = circle_annotator.annotate(
+            ...     scene=image.copy(),
+            ...     detections=detections
+            ... )
+            ```
```
Comment (SkalskiP) [inline supervision/annotators/core.py:368]: Could you add this link to an example annotated frame in docs? `https://media.roboflow.com/supervision-annotator-examples/circle-annotator-example.png` In a similar way, we have done it for other annotators.

**Review by SkalskiP — CHANGES_REQUESTED** [review]

**Comment by SkalskiP** [comment]
> Habits :=) I will use here more okey 👍 :)

Cool. One comment, and we can merge.

File: supervision/annotators/core.py (hunk 1 above)
Comment (onuralpszr) [reply supervision/annotators/core.py:368]: @SkalskiP  I was going to ask that, awesome thank you !!

Comment (SkalskiP) [reply supervision/annotators/core.py:368]: No worries ;) let me know once you update it. ;)

**Comment by onuralpszr** [comment]
@SkalskiP  all done you can merge

**Comment by SkalskiP** [comment]
One more thing. I see that you marked it as Hacktoberfest task. I'm all in favor. Small question: do we need to do anything else? Do you need issue associated with the PR? Is there anything else we need to do to make that PR count to Hacktoberfest?

**Comment by onuralpszr** [comment]
> @onuralpszr One more thing. I see that you marked it as Hacktoberfest task. I'm all in favor. Small question: do we need to do anything else? Do you need issue associated with the PR? Is there anything else we need to do to make that PR count to Hacktoberfest?

"hacktoberfest-accepted" is enough to make it count.

**Comment by SkalskiP** [comment]
Awesome! In that case, merging! 🚀

**Review by SkalskiP — APPROVED** [review]

**Pull request merged**